src/applications/check-in/components/tests/ActionItemDisplay.unit.spec.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { expect } from 'chai';
+import { render } from '@testing-library/react';
+import sinon from 'sinon';
+import CheckInProvider from '../../tests/unit/utils/CheckInProvider';
+import {
+  multipleAppointments,
+  singleAppointment,
+} from '../../tests/unit/mocks/mock-appointments';
+import ActionItemDisplay from '../ActionItemDisplay';
+import * as appointmentModule from '../../utils/appointment';
+
+describe('unified check-in experience', () => {
+  describe('ActionItemDisplay', () => {
+    it('displays the what to do next component if the app is day of', () => {
+      const mockstore = {
+        upcomingAppointments: multipleAppointments,
+        app: 'dayOf',
+      };
+      const { getByTestId } = render(
+        <CheckInProvider store={mockstore}>
+          <ActionItemDisplay />
+        </CheckInProvider>,
+      );
+      expect(getByTestId('what-to-do-next')).to.exist;
+    });
+    it('displays the what to do next component if the app is day of and pre check in is complete', () => {
+      const mockstore = {
+        upcomingAppointments: singleAppointment,
+        app: 'dayOf',
+      };
+      const preCheckinAlreadyCompletedStub = sinon
+        .stub(appointmentModule, 'preCheckinAlreadyCompleted')
+        .returns(true);
+      const { getByTestId } = render(
+        <CheckInProvider store={mockstore}>
+          <ActionItemDisplay />
+        </CheckInProvider>,
+      );
+
+      expect(getByTestId('what-to-do-next')).to.exist;
```

Comment:
I don't feel like this is fully testing what it says. I think you also need to assert that the appointment is already completed.

Reply:
This is testing to make sure it shows the `WhatToDoNext` component on the day-of app regardless of the pre-check-in demographics status.